### PR TITLE
feat: V1

### PR DESCRIPTION
**Pull request by Evalir**
V1 working branch. Includes
- `foundryup` v1 changes:
  - ability to install latest stable release through `foundryup`
  - ability to install last nightly through `--legacy`
  - ability to install an arbitrary stable version through `--version`
- `expect` cheatcode changes. Emit, Call, and Revert all now work for the next call, instead of test level.

### CODE REVIEW

File: foundryup/foundryup [hunk 1]
```diff
@@ -24,6 +25,7 @@ main() {
       -p|--path)        shift; FOUNDRYUP_LOCAL_REPO=$1;;
       -P|--pr)          shift; FOUNDRYUP_PR=$1;;
       -C|--commit)      shift; FOUNDRYUP_COMMIT=$1;;
+      -L|--legacy)      shift; FOUNDRYUP_LEGACY=true;;
```
Comment (gakonst) [inline foundryup/foundryup:28]: Good!

File: testdata/cheats/Etch.t.sol [hunk 2]
```diff
@@ -14,12 +14,12 @@ contract EtchTest is DSTest {
...
 
-    function testEtchNotAvailableOnPrecompiles() public {
-        address target = address(1);
-        bytes memory code = hex"1010";
-        vm.expectRevert(
-            bytes("Etch cannot be used on precompile addresses (N < 10). Please use an address bigger than 10 instead")
-        );
-        vm.etch(target, code);
-    }
+    // function testEtchNotAvailableOnPrecompiles() public {
```
Comment (gakonst) [inline testdata/cheats/Etch.t.sol:17]: What's up

**Review by gakonst — APPROVED** [review]
Cc @mattsse lgtm pls do sanity pass at the sol tests UX

File: testdata/cheats/Etch.t.sol (hunk 2 above)
Comment (Evalir) [reply testdata/cheats/Etch.t.sol]: Ah, commented by mistake when making the PR. fix https://github.com/foundry-rs/foundry/pull/5223/commits/1e4f206afd4cb2bc7b301ef4bba7bf2b666fb331

**Pull request merged**